File: Data Structures & Algorithms/validate-parentheses/submission-0.js
class Solution {
    /**
     * @param {string} s
     * @return {boolean}
     */
    isValid(s) {
        const match = {
            '}' : '{',
            ']' : '[',
            ')' : '('
        }

        const stack = []
        for(let char of s){
            if(match[char]){
                const present = stack.pop();
                if(present!==match[char]) return false
            } else{
                stack.push(char)
            }
        }
        return stack.length!==0 ? false : true 
    }

}

File: Data Structures & Algorithms/decode-string/submission-0.py
class Solution:
    def decodeString(self, s: str) -> str:
        stack = []
        ans = ''

        for let in s:
            if let!=']':
                stack.append(let)
            else:
                tempStr = []
                while stack and stack[-1]!='[':
                    tempStr.append(stack.pop())
                stack.pop()

                k = ''
                while stack and stack[-1].isdigit():
                    k = stack.pop()+k
                finalStr = int(k)*"".join(reversed(tempStr))
                stack.append(finalStr)
        return "".join(stack)